// @react-ui/atoms-css-modules/lib/components/form/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _form = require('./form');

var _form2 = _interopRequireDefault(_form);

var _group = require('./group');

var _group2 = _interopRequireDefault(_group);

var _input = require('./input');

var _input2 = _interopRequireDefault(_input);

var _label = require('./label');

var _label2 = _interopRequireDefault(_label);

var _textarea = require('./textarea');

var _textarea2 = _interopRequireDefault(_textarea);

var _select = require('./select');

var _select2 = _interopRequireDefault(_select);

var _static = require('./static');

var _static2 = _interopRequireDefault(_static);

var _helper = require('./helper');

var _helper2 = _interopRequireDefault(_helper);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Form = {
  Container: _form2.default,
  Group: _group2.default,
  Input: _input2.default,
  Label: _label2.default,
  Textarea: _textarea2.default,
  Select: _select2.default,
  Static: _static2.default,
  Helper: _helper2.default
};

exports.default = Form;
