// @react-ui/atoms-css-modules/lib/components/loading/loading.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Loading = function Loading() {
  var styles = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];

  return function (_Component) {
    _inherits(LoadingInner, _Component);

    function LoadingInner(props) {
      _classCallCheck(this, LoadingInner);

      var _this = _possibleConstructorReturn(this, Object.getPrototypeOf(LoadingInner).call(this, props));

      _this.state = {
        loading: false
      };
      return _this;
    }

    _createClass(LoadingInner, [{
      key: 'componentWillMount',
      value: function componentWillMount() {
        var _this2 = this;

        this.timeoutId = setTimeout(function () {
          _this2.setState({ loading: true });
        }, 300);
      }
    }, {
      key: 'componentWillUnmount',
      value: function componentWillUnmount() {
        clearTimeout(this.timeoutId);
      }
    }, {
      key: 'render',
      value: function render() {
        var _props = this.props;
        var className = _props.className;

        var rest = _objectWithoutProperties(_props, ['className']);

        var activeClass = _defineProperty({}, styles.active, !!this.state.loading);

        return _react2.default.createElement(
          'div',
          _extends({}, rest, { className: (0, _classnames2.default)(styles.loading, activeClass, className) }),
          _react2.default.createElement(
            'div',
            { className: styles.element },
            _react2.default.createElement('div', { className: styles.item }),
            _react2.default.createElement('div', { className: styles.item }),
            _react2.default.createElement('div', { className: styles.item })
          )
        );
      }
    }]);

    return LoadingInner;
  }(_react.Component);
};

exports.default = Loading;
